components/channel_header_mobile/channel_header_mobile.js

```diff
@@ -90,29 +90,36 @@ export default class ChannelHeaderMobile extends React.PureComponent {
                 <div className='container-fluid theme'>
                     <div className='navbar-header'>
                         <CollapseLhsButton/>
+                        {channel ? (
+                            <div className='navbar-brand'>
+                                <MobileChannelHeaderDropdown/>
+                                {isMuted && (
+                                    <UnmuteChannelButton
+                                        user={user}
+                                        channel={channel}
+                                    />
+                                )}
+                            </div>
+                        ) : (
+                            <div className='navbar-brand'>
+                                {'Threads'}
+                            </div>
+                        )}
+                        <div className='spacer'/>
+                        {channel && (
+                            <ChannelInfoButton
+                                channel={channel}
+                                isReadOnly={isReadOnly}
+                                isRHSOpen={isRHSOpen}
+                                currentRelativeTeamUrl={currentRelativeTeamUrl}
+                            />
+                        )}
+                        <ShowSearchButton/>
                         {channel && (
-                            <React.Fragment>
-                                <div className='navbar-brand'>
-                                    <MobileChannelHeaderDropdown/>
-                                    {isMuted && (
-                                        <UnmuteChannelButton
-                                            user={user}
-                                            channel={channel}
-                                        />
-                                    )}
-                                </div>
-                                <ChannelInfoButton
-                                    channel={channel}
-                                    isReadOnly={isReadOnly}
-                                    isRHSOpen={isRHSOpen}
-                                    currentRelativeTeamUrl={currentRelativeTeamUrl}
-                                />
-                                <ShowSearchButton/>
-                                <MobileChannelHeaderPlug
-                                    channel={channel}
-                                    isDropdown={false}
-                                />
-                            </React.Fragment>
+                            <MobileChannelHeaderPlug
+                                channel={channel}
+                                isDropdown={false}
+                            />
```

Comment:
I'm using inline conditionals here to avoid two very similar trees with would-be duplicated content. It would be more straight forward if the items weren't interwoven, and we should probably improve this, but I was trying to reduce changes to core layout when possible.